app/parsing_scripts/git_update.py

```diff
@@ -18,54 +18,50 @@
 
 
 def clone_repo():
-    if os.path.isdir(config.REPO):
+    os.chdir(config.DATA_DIR)
+    if os.path.isdir(config.MACPORTS_PORTS_DIR):
         return
     else:
-        os.system("git clone %s" % config.REPO_URL)
+        subprocess.run([config.GIT, 'clone', config.MACPORTS_PORTS_URL])
 
 
-def get_list_of_changed_ports(new_hash=False, old_hash=False, root=BASE_DIR):
-    os.chdir(root)
+def get_list_of_changed_ports(new_hash=False, old_hash=False):
+    if os.path.isdir(config.MACPORTS_PORTS_DIR):
```

Comment:
This is a bit of nitpicking, but when code consists of a super log portion with the main loop + a tiny bit of code to handle the `else` part, it's more difficult to follow the flow (and even find the `else` part). What you can do is start with the short else part and immediately return. Then for the main part you don't even need to use extra alignment. I can tell more in a chat if needed.